power_grid_model_c/power_grid_model/include/power_grid_model/common/calculation_info.hpp

```diff
@@ -4,15 +4,75 @@
 
 #pragma once
 
-#include "logging.hpp"
+#include "logging_impl.hpp"
 
-#include <cstddef>
-#include <functional>
 #include <map>
-#include <string>
 
 namespace power_grid_model {
+namespace common::logging {
+class CalculationInfo : public NoLogger {
+    using Data = std::map<LogEvent, double>;
 
-using CalculationInfo = std::map<LogEvent, double>;
+  public:
+    using Report = std::add_lvalue_reference_t<std::add_const_t<Data>>;
+    static_assert(std::same_as<Report, Data const&>);
 
+    void log(LogEvent tag, double value) override { log_impl(tag, value); }
+    void log(LogEvent tag, Idx value) override { log_impl(tag, static_cast<double>(value)); }
```

Comment:
Should we add a super generic `log(LogEvent tag)` without value to just log the event taking place? 
```suggestion
    void log(LogEvent tag, double value) override { log_impl(tag, value); }
    void log(LogEvent tag, Idx value) override { log_impl(tag, static_cast<double>(value)); }
    void log(LogEvent tag) override { log_impl(tag); }
```

Reply:
apologies. i did add this but apparently didn't commit. reopening

Reply:
OHHH i see, i did add it to the logger interface but not here.

What should the report look like? integer counter (similar to how `maximum_num_iterations` is a double but represents a float)?

Reply:
thinking about it, it is tricky because it depends on the event whether it should time as `0`, or keep track of a counter... maybe it's good to add it when the first event comes into play that should just be an event counter

Reply:
that would be a full-fledged logger, which has its own purpose (possible future addition). this calculation info is not a full-fledged logger but an aggregate, so indeed `vector` it is, then

Reply:
With `vector` one could still record multiple occurrences of single type event. If we stick to the map and give it value `0`, it will simple keep rewriting `0` with `0`.

Reply:
thinking about it, i really don't think we should add this right now unless there is a specific event we want to capture this way (we currently do not, but if you have a specific request, please LMK)

as you mention, indeed, `0` as a default value is a bad value, but it will add runtime overhead when having to check `is_nan` for every log, so even adding it to the map is not a good idea in the first place.

alternatively, a separate vector means that the report will have to be created (copied) at report time and passed by value, not by reference. While i'm ok with that, there's no need to build that now.

in addition, how should the reporting happen when there are multiple events in the vector, and also events listed in the other data.

all things considered, it feels like too much implementation details even though there's no concrete use case of it.